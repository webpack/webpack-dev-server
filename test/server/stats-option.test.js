'use strict';

const webpack = require('webpack');
const Server = require('../../lib/Server');
const config = require('../fixtures/simple-config/webpack.config');
const port = require('../ports-map')['stats-option'];

describe('stats option', () => {
  it(`should works with difference stats values (contains 'hash', 'assets', 'warnings' and 'errors')`, () => {
    const allStats = [
      {},
      // eslint-disable-next-line no-undefined
      undefined,
      false,
      'errors-only',
      {
        assets: false,
      },
    ];

    return allStats.reduce((p, stats) => {
      return p.then(() => {
        return new Promise((resolve) => {
          const compiler = webpack(config);
          const server = new Server(compiler, { stats, port, quiet: true });

          compiler.hooks.done.tap('webpack-dev-server', (s) => {
            expect(Object.keys(server.getStats(s))).toMatchSnapshot();

            server.close(resolve);
          });

          compiler.run(() => {});
          server.listen(port, 'localhost');
        });
      });
    }, Promise.resolve());
  });

  it('should respect warningsFilter', (done) => {
    const compiler = webpack(config);
    const server = new Server(compiler, {
      stats: { warningsFilter: 'test' },
      port,
      quiet: true,
    });

    compiler.hooks.done.tap('webpack-dev-server', (s) => {
      s.compilation.warnings = ['test', 'another warning'];

      const output = server.getStats(s);

      expect(output.warnings.length).toBe(1);
      expect(output.warnings[0]).toBe('another warning');

      server.close(done);
    });

    compiler.run(() => {});
    server.listen(port, 'localhost');
  });
});
