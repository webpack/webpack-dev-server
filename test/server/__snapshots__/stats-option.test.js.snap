// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stats option should works with difference stats values (contains 'hash', 'assets', 'warnings' and 'errors') 1`] = `
Array [
  "errors",
  "warnings",
  "hash",
  "assetsByChunkName",
  "assets",
  "filteredAssets",
]
`;

exports[`stats option should works with difference stats values (contains 'hash', 'assets', 'warnings' and 'errors') 2`] = `
Array [
  "errors",
  "warnings",
  "hash",
  "assetsByChunkName",
  "assets",
  "filteredAssets",
]
`;

exports[`stats option should works with difference stats values (contains 'hash', 'assets', 'warnings' and 'errors') 3`] = `
Array [
  "errors",
  "warnings",
  "hash",
  "assetsByChunkName",
  "assets",
  "filteredAssets",
]
`;

exports[`stats option should works with difference stats values (contains 'hash', 'assets', 'warnings' and 'errors') 4`] = `
Array [
  "errors",
  "warnings",
  "hash",
  "assetsByChunkName",
  "assets",
  "filteredAssets",
]
`;

exports[`stats option should works with difference stats values (contains 'hash', 'assets', 'warnings' and 'errors') 5`] = `
Array [
  "errors",
  "warnings",
  "hash",
  "assetsByChunkName",
  "assets",
  "filteredAssets",
]
`;
